app-creator/code-editor/deserializer.js

```diff
@@ -0,0 +1,423 @@
+var mapStyles = require('users/msibrahim/app-creator:map-styles');
+var examples = require('users/msibrahim/app-creator:examples');
+
+exports.createApp = createApp;
+
+function createApp(template) {
+  /**
+   * The namespace for the application. All the state is kept in here.
+   */
+  var app = {};
+
+  /**
+   * Allow users to further customize the generated widgets by providing a reference to each element created.
+   * Type: {[key: WidgetID]: {node: EEWidget, map?: ui.Map}}, for map widgets, we wrap them with a panel widget and return
+   * both elements.
+   */
+  app.widgetInterface = ui.data.ActiveDictionary();
+
+  /**
+   * Returns reference to the widget interface.
+   */
+  app.widgets = function () {
+    return app.widgetInterface;
+  };
+
+  /**
+   * Recursively traverses the widget tree starting at panel-template-0 and creates ee ui widgets accordingly.
+   */
+  app.deserializeUI = function (widgetTreeJSON) {
+    function helper(nodeObj) {
+      var obj = app.createUIElement(nodeObj);
+      if (!obj) {
+        console.log('Incorrect widget type');
```

Comment:
Use print instead of console.log, perhaps also print out the node object that's failing too:
`print('Received invalid node object', nodeObj);`